include/ddc/kernels/splines/matrix_sparse.hpp

```diff
@@ -116,91 +90,39 @@ unsigned int default_preconditionner_max_block_size() noexcept
     return 1u;
 }
 
-// Matrix class for Csr storage and iterative solve
+// Matrix class for sparse storage and iterative solve
 template <class ExecSpace>
 class Matrix_Sparse : public Matrix
 {
-private:
-    Kokkos::View<int*, Kokkos::HostSpace> m_rows;
+    using matrix_sparse_type = gko::matrix::Csr<double, int>;
 
-    Kokkos::View<int*, Kokkos::HostSpace> m_cols;
+private:
+    std::unique_ptr<gko::matrix::Dense<double>> m_matrix_dense;
 
-    Kokkos::View<double*, Kokkos::HostSpace> m_data;
+    std::shared_ptr<matrix_sparse_type> m_matrix_sparse;
 
-    std::unique_ptr<gko::solver::Bicgstab<double>::Factory> m_solver_factory;
+    std::shared_ptr<gko::solver::Bicgstab<double>> m_solver;
 
     int m_cols_per_par_chunk; // Maximum number of columns of B to be passed to a Ginkgo solver
```

Comment:
Should I rename this as `m_chunk_size` everywhere ?

Reply:
Yes good idea